mysql-exporter/mysql-exporter.libsonnet

```diff
@@ -3,38 +3,61 @@ local container = k.core.v1.container;
 local service = k.core.v1.service;
 local containerPort = k.core.v1.containerPort;
 local deployment = k.apps.v1.deployment;
+local envVar = k.core.v1.envVar;
+
+
+local mysql_credential(config) =
+  if std.length(config.mysql_password) > 0 then
+    [{ name: 'MYSQL_PASSWORD', value: config.mysql_password }]
+  else [envVar.fromSecretRef('MYSQL_PASSWORD', config.mysql_password_secret, config.mysql_password_secret_key)];
```

Comment:
With the env dict, you can now create a mixin function like this that would replace this `mysql_credentials()` function.

```
  withSecretPassword(name, key='password'):: {
    env+:: {
      MYSQL_PASSWORD:: '',
    },
    mysqld_exporter_container+::
      container.withEnvMixin([
        envVar.fromSecretRef('MYSQL_PASSWORD', name, key),
      ]),
  },
```